.github/workflows/ci.yml

```
@@ -12,7 +12,7 @@ jobs:
       - name: Install Go
         uses: actions/setup-go@v2
         with:
-          go-version: "1.17.x"
+          go-version: "1.22.1"
```

```suggestion
          go-version-file: "go.mod"
```

Idea: I think this will make synchronizing the versions in the future easier.
https://github.com/actions/setup-go?tab=readme-ov-file#getting-go-version-from-the-gomod-file